COLLABORATOR_GUIDE.md

```diff
@@ -0,0 +1,96 @@
+# jade Collaborator Guide
```

Comment:
Well there is a talk in `io.js` about moving those files into separate directory (or a separate repository even). Relevant discussion is here: https://github.com/nodejs/node/pull/12